src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -32,6 +32,17 @@
 
 class PythonAwsLambdaSources(PythonSources):
     expected_num_files = range(0, 2)
+    deprecated_removal_version = "2.3.0.dev0"
+    deprecated_removal_hint = (
+        "Remove the `sources` field and create a `python_library` target with the handler "
+        "file included (if it does not yet exist). Pants will infer a dependency, which you can "
+        "check with `./pants dependencies path/to:lambda`. See "
+        "https://www.pantsbuild.org/v2.2/docs/awslambda-python for an example.\n\nYou can also "
```

Comment:
Benjy was originally concerned that we can only have 5 versions of docs on Readme. We now have >5, so it seems that it's fine.

I'm not worried about this becoming stale because it's a deprecation message so will be deleted in 2.3.